pandas/core/groupby/generic.py

```diff
@@ -259,7 +259,13 @@ def aggregate(self, func=None, *args, **kwargs):
                 result = self._aggregate_named(func, *args, **kwargs)
 
             index = Index(sorted(result), name=self.grouper.names[0])
-            ret = Series(result, index=index)
+
```

Comment:
I think you can create a function to do this rather than have if/elses in multiple places. put it in pandas/core/series.py is ok.